### PR TITLE
Add function to load PREM

### LINKED ISSUE #8: PREM
**Description of the desired feature**

Include PREM (http://ds.iris.edu/ds/products/emc-prem/). This should be relatively straight forward as the model can be downloaded in CSV format and quickly loaded with pandas.

### PR DESCRIPTION
**Pull request by santisoler**
Add function to load PREM data into `pandas.DataFrame`.
Add tests for the new function, entry in the documentation and a gallery example.
Add `pandas` as dependency.

Fixes #8 

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [x] Add new public functions/methods/classes to `doc/api/index.rst`.
- [x] Write detailed docstrings for all functions/methods.
- [x] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.

### CODE REVIEW

**Comment by santisoler** [comment]
@leouieda 
I have a few questions:
* Should we add `pandas` as dependency?
* Once the documentation is build, the `pandas.DataFrame` has no link, how can we solve it?
* `pylint` raises an error with the `:math:\eta` string in the docstring of `fetch_prem()` function. I don't know why this would be an error.

File: rockhound/prem.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    Parameters
+    ----------
+    load : bool
+        Wether to load the data into a :class:`pandas.DataFrame` or just return the
+        path to the downloaded data.
+
+    Returns
+    -------
+    prem : :class:`pandas.DataFrame` or str
+        The loaded data or the file path ot the downloaded data.
```
Comment (leouieda) [inline rockhound/prem.py]: Could you please add the names and units of the columns in the DataFrame?

File: rockhound/prem.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Load the Preliminary Reference Earth Model (PREM) dataset.
+"""
+import numpy as np
+import pandas as pd
+from .registry import REGISTRY
+
+
+def fetch_prem(load=True):
+    """
+    Fetch the Preliminary Reference Earth Model (PREM) model.
```
Comment (leouieda) [inline rockhound/prem.py]:
```suggestion
    Fetch the Preliminary Reference Earth Model (PREM).
```

File: rockhound/prem.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+"""
+Load the Preliminary Reference Earth Model (PREM) dataset.
+"""
+import numpy as np
+import pandas as pd
+from .registry import REGISTRY
+
+
+def fetch_prem(load=True):
+    """
```
Comment (leouieda) [inline rockhound/prem.py]:
```suggestion
    r"""
```

You need to make this a raw string in order to have `\` in there. I think that's what pylint is complaining about.

**Review by leouieda — CHANGES_REQUESTED** [review]
> Should we add pandas as dependency?

Yes please. Add it to `requirements.txt`, `setup.py`, `doc/install.rst` and `environment.yml`.

> Once the documentation is build, the pandas.DataFrame has no link, how can we solve it?

That's done by intersphinx. You'll need to add this line: https://github.com/fatiando/harmonica/blob/master/doc/conf.py#L34

**Comment by santisoler** [comment]
@leouieda 
I've added `pandas` as dependency, fixed the `r"""` issue and added columns names a their units in `prem` `DataFrame`.

I've also fixed some issues regarding styling.
If you agree, this PR is ready to merge.

**Review by leouieda — APPROVED** [review]

**Comment by leouieda** [comment]
:+1: thanks, Santi!

**Pull request merged**